ts-client/src/dlmm/index.ts

```diff
@@ -3070,32 +3123,51 @@ export class DLMM {
         tokenYProgram: TOKEN_PROGRAM_ID,
         sender: user,
       })
-      .preInstructions(preInstructions)
-      .postInstructions(postInstructions)
-      .transaction();
+      .instruction();
+
+    const instructions = [
+      ...preInstructions,
+      removeLiquidityIx,
+      ...postInstructions,
+    ];
+
+    const setCUIx = await getEstimatedComputeUnitIxWithBuffer(
+      this.program.provider.connection,
+      instructions,
+      user
+    );
+
+    instructions.unshift(setCUIx);
 
     const { blockhash, lastValidBlockHeight } =
```

Comment:
that should be fetch inside the block if/else, right before return transaction

Reply:
Fixed